src/IoTSettings.ts

```diff
@@ -3,47 +3,36 @@
 
 import * as path from 'path';
 import * as vscode from 'vscode';
-import * as sdk from 'vscode-iot-device-cube-sdk';
 
 import {CancelOperationError} from './CancelOperationError';
 import {ConfigHandler} from './configHandler';
-import {OSPlatform} from './constants';
+import {ConfigKey, OSPlatform} from './constants';
 import {PickWithData} from './Models/Interfaces/UI';
+import {getHomeDir, getPlatform} from './utils';
 
 export class IoTWorkbenchSettings {
-  private _workbenchPath = '';
+  private workbenchPath = '';
+  private static instance: IoTWorkbenchSettings|undefined;
 
   private constructor() {}
 
-  static async createAsync() {
-    const iotWorkbenchSettings = new IoTWorkbenchSettings();
-
-    iotWorkbenchSettings._workbenchPath =
-        await IoTWorkbenchSettings.getWorkbenchPath();
+  static async getInstance() {
+    if (!this.instance) {
+      this.instance = new IoTWorkbenchSettings();
+      this.instance.workbenchPath =
+          ConfigHandler.get<string>(ConfigKey.workbench) ||
+          (await this.getDefaultWorkbenchPath());
+      await ConfigHandler.update(

```

Comment:
Do we really need to update config again if Configkey.workbench is already set? I guess you mean update the config when it is not set?

Reply:
Yes. Update user setting when it is not set.